[ENG-5615] Add multiple integration settings and Additional Api mappings

huzaifa-binafzal opened the pull request.

husseinmimiHarri commented on main.tf:
```diff
@@ -204,7 +221,7 @@ resource "aws_apigatewayv2_route" "this" {
   route_key                           = each.key
   route_response_selection_expression = local.is_websocket ? each.value.route_response_selection_expression : null
-  target                              = "integrations/${aws_apigatewayv2_integration.this[each.key].id}"
+  target                              = "integrations/${aws_apigatewayv2_integration.this[coalesce(each.value.integration.key, each.key)].id}"
```
change each.value.integration.key to each.value.integration_key

husseinmimiHarri commented on variables.tf:
```diff
@@ -244,7 +253,7 @@ variable "routes" {
 
     # Integration
-    integration = object({
+    integration = map(object({
```
we want to have integration attached to multiple routes, we don't need this as map

husseinmimiHarri replied:
i think we need to remove integration map from here and create integrations variable, which is map of integrations then we pass needed integration to the route

husseinmimiHarri commented on variables.tf:
```diff
@@ -105,6 +105,15 @@ variable "api_mapping_key" {
   default     = null
 }
 
+variable "additional_default_stage_api_mappings" {
```
lets change this to additional_stage_api_mapping because we can name created stage now

husseinmimiHarri commented on main.tf:
```diff
@@ -116,6 +122,17 @@ resource "aws_apigatewayv2_api_mapping" "this" {
+
+  api_id          = aws_apigatewayv2_api.this[0].id
+  domain_name     = each.value["domain_name"]
+  stage           = aws_apigatewayv2_stage.default[0].id
```
aws_apigatewayv2_stage.this[0].id

huzaifa-binafzal commented:
Changes made to module:

- Added support for integration to accept null value for the case where the route doesn't have a integration attached.

- Added route_setting variable to fix the issue were it always created rout_setting even if it existed. And to support more then 1 rout_settings
